lib/zk_recipes/cache.rb

```diff
@@ -118,22 +204,66 @@ def reopen
       end
     end
 
+    # Fetch the parsed ZK value or default value if the ZK value is missing or corrupt.
+    #
+    # This method is threadsafe.
+    #
+    # @param path [String]
+    # @return [Object] The parsed value for `path` or the default value if `path` is not set or there was a
+    # deserialization error.
     def fetch(path)
-      @cache.fetch(path).value
-    rescue KeyError
-      raise PathError, "no registered path for #{path.inspect}"
+      static_path = @static_paths[path]
+      raise PathError, "no registered path for #{path.inspect}" unless static_path
+
+      static_path.value
     end
     alias_method :[], :fetch
 
+    # Fetch valid parsed ZK value only, do not return the default value if the ZK value is missing or corrupt.
+    #
+    # This method is threadsafe.
+    #
+    # @param path [String]
+    # @return [Object] The parsed value for `path` or `nil` if `path` is not set or there was a deserialization error.
     def fetch_valid(path)
-      cached = @cache.fetch(path)
-      cached.value if cached.valid?
-    rescue KeyError
-      raise PathError, "no registered path=#{path.inspect}"
+      static_path = @static_paths[path]
+      raise PathError, "no registered path for #{path.inspect}" unless static_path
+
+      static_path.valid_value
+    end
+
+    # Fetch the values for each mapped child of `path`.
+    #
+    # This method is threadsafe.
+    #
+    # @param path [String] The directory path.
+    # @return [Hash<String, Object>] Hash of parsed values for each child's mapped path in the directory.
+    #
+    # Mapped paths that:
+    # - don't exist and aren't static registered paths (see `#register`) are not included.
+    # - do exist and aren't static registered paths but had deserialization errors are not included.
+    # - are static registered paths are included and will be the parsed or default value.
+    #
+    # Note: if multiple registered directories have children that map to the same path and that path is not statically
+    # registered, each directory's deserializer will be used to parse that path's value.
+    #
+    # See {file:examples/cache.rb}.
```

Comment:
YARDoc nits:

* `@return` is typically at the bottom of a comment block
* `@note` and `@see` are also annotations you can use

Reply:
If you have extended comments on the `@return` value, should that go above or below the `@return` annotation?